lib/spack/llnl/util/tty/log.py

```diff
@@ -449,7 +449,7 @@ def _writer_daemon(self, stdin):
                     # Without the last parameter (timeout) select will
                     # wait until at least one of the two streams are
                     # ready. This may cause the function to hang.
```

Comment:
How do you cope with the three lines of comments above? Is there any guarantee we won't hang?

Reply:
In fact, I've done some testing on Linux (Centos 6 and Ubuntu 18.04) and I do not understand how the routine can hang.
If there is something to read either on the pipe or stdin, select will return.
If there is nothing to read, that means there is nothing to do and the process is blocked on select, until there is something to do. So this is the case where it can hangs. On Linux, closing file descriptor causes select to exit with an exception, so if the pipe is broken, this function does not hang.

By looking at the history, it seems that the '0' was introduced by @tgamblin in commit https://github.com/spack/spack/commit/36a4ca8b113056b20435cd38b00551c1c9275def but at the time, there was no test if readline was returning something. Now EOF is taking care of by the break below (line 467)

I do not have a Mac so I cannot test if it can work properly on a non linux system.